galsim/catalog.py

```diff
@@ -343,3 +349,255 @@ def __hash__(self): return hash(repr(self))
 
 
 
+class OutputCatalog(object):
+    """A class for building up a catalog for output, typically storing truth information
+    about a simulation.
+
+    Each row corresponds to a different object, and each column stores some item of
+    information about that object (e.g. flux or half_light_radius).
+
+    Note: no type checking is done when the data are added in add_row().  It is up to 
+    the user to make sure that the values added for each row are compatible with the
+    types given here in the `types` parameter.
+
+    Initialization
+    --------------
+
+    @param names    A list of names for the output columns.
+    @param types    A list of types for the output columns. [default: None, which assumes all
+                    columns are float]
+
+    Attributes
+    ----------
+
+    After construction, the following attributes are available:
+
+        nobjects    The number of objects so far in the catalog.
+        ncols       The number of columns in the catalog.
+        names       The names of the columns.
+        types       The types of the columns.
+        cols        The columns of data that have been accumulated so far.
+
+    """
+    # Watch out for this "Gotcha".  Using _rows=[] as the default argument doesn't work!
+    # https://pythonconquerstheuniverse.wordpress.com/2012/02/15/mutable-default-arguments/
+    def __init__(self, names, types=None, _rows=None, _sort_keys=None):
+        self.names = names
+        if types is None:
+            self.types = [ float for i in names ]
+        else:
+            self.types = types
+        if _rows is None:
+            self.rows = []
+        else:
+            self.rows = _rows
+        if _sort_keys is None:
+            self.sort_keys = []
+        else:
+            self.sort_keys = _sort_keys
+
+    @property
+    def nobjects(self): return len(self.rows)
+    @property
+    def ncols(self): return len(self.names)
+
+    # Again, when we use this through a proxy, we need getters for the attributes.
+    def getNames(self): return self.names
+    def getTypes(self): return self.types
+    def setTypes(self, types): self.types = types
+    def getNObjects(self): return self.nobjects
+    def getNCols(self): return self.ncols
+
+    def add_row(self, row, sort_key=None):
```

Comment:
I see a mix of camelcase and underscore+lower method names here.  Is this intentional, to be consistent with similar method names elsewhere?